src/CMakeLists.txt

```diff
@@ -871,16 +871,14 @@ add_subdirectory(compressor)
 
 add_subdirectory(tools)
 
-# dmClock
-
-add_subdirectory(dmclock) # after gmock
-add_dependencies(tests dmclock-tests dmclock-data-struct-tests)
+# dmClock (after gmock)
 
+add_subdirectory(dmclock/src)
 if(WITH_TESTS)
-  install(PROGRAMS
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-tests
-    ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/dmclock-data-struct-tests
-    DESTINATION bin)
+  # note: add_test is not being called, so dmclock tests aren't part
+  # of ceph tests
+  add_subdirectory(dmclock/test)
```

Comment:
That's the whole problem - WITH_TESTS is true by default. See https://github.com/ceph/ceph/blob/master/CMakeLists.txt#L453

(Actually, I need the dmclock tests to not be built regardless of the setting of WITH_TESTS. That's why I'm asking you to change this conditional to use something else, like WITH_DMCLOCK_TESTS. Thanks for considering this request.)